cluster-operator/src/main/java/io/strimzi/operator/cluster/model/RackConfig.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.model;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class RackConfig {
+
+    private String topologyKey;
+
+    public RackConfig() {
+
+    }
+
+    public RackConfig(String topologyKey) {
+        this.topologyKey = topologyKey;
+    }
+
+    @JsonProperty("topologyKey")
+    public String getTopologyKey() {
+        return topologyKey;
+    }
+
+    public static RackConfig fromJson(String json) {
+        RackConfig rackConfig = JsonUtils.fromJson(json, RackConfig.class);
+        if (rackConfig != null && (rackConfig.getTopologyKey() == null || rackConfig.getTopologyKey().equals(""))) {
+            throw new IllegalArgumentException("In rack configuration the 'topologyKey' field is mandatory");
```

Comment:
Shame we can't really use `InvalidConfigMapException` here, because we don't know the key (indeed we don't know the given `json` was from a CM at all): This means we've lost info about which key is the problem. I know in this case it's not really ambiguous, because there's only one rack config, but it suggests we need a more sophistocated way of coping with errors like this.